drivers/volume/portworx/portworx.go

```diff
@@ -4170,6 +4178,141 @@ func (d *portworx) RecoverNode(n *node.Node) error {
 
 }
 
+// AddBlockDrives add drives to the node using PXCTL
+func (d *portworx) AddBlockDrives(n *node.Node, drivePath []string) error {
+
+	systemOpts := node.SystemctlOpts{
+		ConnectionOpts: node.ConnectionOpts{
+			Timeout:         startDriverTimeout,
+			TimeBeforeRetry: defaultRetryInterval,
+		},
+		Action: "start",
+	}
+	logrus.Infof("Getting available block drives on %s.", n.Name)
+	blockDrives, err := d.nodeDriver.GetAvailableDrives(*n, systemOpts)
+
+	if err != nil {
+		return err
+	}
+
+	eligibleDrives := make(map[string]*node.BlockDrive)
+
+	for _, drv := range blockDrives {
+		if drv.MountPoint == "" && drv.FSType == "" && drv.Type == "disk" {
+			eligibleDrives[drv.Path] = drv
+		}
+	}
+
+	if len(blockDrives) == 0 || len(eligibleDrives) == 0 {
+		return fmt.Errorf("no block drives available to add")
+	}
+
+	if drivePath == nil || len(drivePath) == 0 {
+		logrus.Infof("Adding all the available drives")
+		for _, drv := range eligibleDrives {
+			err := addDrive(*n, drv.Path, d)
+			if err != nil {
+				return err
+			}
+			err = waitForAddDriveToComplete(*n, drv.Path, d)
+			if err != nil {
+				return err
+			}
+		}
+	} else {
+		for _, drvPath := range drivePath {
+			drv, ok := eligibleDrives[drvPath]
+			if ok {
+				err := addDrive(*n, drv.Path, d)
+				if err != nil {
+					return err
+				}
+				err = waitForAddDriveToComplete(*n, drv.Path, d)
+				if err != nil {
+					return err
+				}
+			} else {
+				return fmt.Errorf("block drive path %s is not eligible or does not exist to perform drive add", drvPath)
+			}
+
+		}
+	}
+	return nil
+}
+
+func addDrive(n node.Node, drivePath string, d *portworx) error {
+
+	out, err := d.nodeDriver.RunCommandWithNoRetry(n, fmt.Sprintf(pxctlDriveAddStart, d.getPxctlPath(n), drivePath), node.ConnectionOpts{
+		Timeout:         crashDriverTimeout,
+		TimeBeforeRetry: defaultRetryInterval,
+	})
+	if err != nil {
+		err = fmt.Errorf("error when adding drive %s in node %s using PXCTL, Cause: %v", drivePath, n.Name, err)
+		return err
+	}
+	output := []byte(out)
+	var addDriveStatus statusJSON
+	err = json.Unmarshal(output, &addDriveStatus)
+	if err != nil {
+		return fmt.Errorf("ERROR of unmarshal adding drive: %v", err)
+	}
+	if &addDriveStatus == nil {
+		return fmt.Errorf("failed to add drive %s in node %s", drivePath, n.Name)
+	}
+
+	if !strings.Contains(addDriveStatus.Status, "Drive add done") {
+		return fmt.Errorf("failed to add drive %s in node %s,AddDrive Status : %+v ", drivePath, n.Name, addDriveStatus)
+
+	}
+	logrus.Infof("Added drive %s to node %s successfully", drivePath, n.Name)
+
+	return nil
+
+}
+
+func waitForAddDriveToComplete(n node.Node, drivePath string, d *portworx) error {
+
+	waitCount := 40
```

Comment:
It would be running rebalance and it again depends on size of data already present on pool. Would 20 minutes be suffice for big pools?

Reply:
Updated to 3 hours